### ISSUE TITLE
(aws-events): `Match.exists()` and `Match.doesNotExist()` is swapped out

### ISSUE THREAD

**Issue by AlessandroVol23** [aws/aws-cdk#23469]
### Describe the issue

I think the documentation of both functions is somehow swapped.

Exists [documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_events.Match.html#static-exists) explains:
```
Matches when the field is absent from the JSON of the event.
```

Doesn't exist [documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_events.Match.html#static-doeswbrnotwbrexist) explains:

```
Matches when the field is present in the JSON of the event.
```

Either I get the functions completely wrong or the docs are wrong.

Also from the [code](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-events/lib/event-pattern.ts#L26)

```ts
  /**
   * Matches when the field is absent from the JSON of the event
   */
  public static exists(): string[] {
    return this.fromObjects([{ exists: true }]);
  }

  /**
   * Matches when the field is present in the JSON of the event
   */
  public static doesNotExist(): string[] {
    return this.fromObjects([{ exists: false }]);
  }
```

I understand it should be vice-versa.

### Links

Does not exist: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_events.Match.html#static-doeswbrnotwbrexist

Exist: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_events.Match.html#static-exists

**Comment by peterwoodworth** [comment]
Yeah, these should be swapped. Good catch! Thanks for calling it out @AlessandroVol23. Would you like to submit a PR to fix this yourself?

**Comment by HBobertz** [comment]
If it's ok I'd like to pick this up 👍

**Comment by peterwoodworth** [comment]
Go for it @HBobertz !

**Comment by AlessandroVol23** [comment]
> If it's ok I'd like to pick this up 👍

Yes please do that :)! I checked it out briefly but I don't have the repo running locally so it would take me some more time 😉

Thanks!

**Comment by HBobertz** [comment]
When I went to pick this up I thought there was an actual logic change to this 😅. Regardless change is out

**Comment by peterwoodworth** [comment]
@HBobertz great, thanks! Be sure to remember to have `fixes` or `closes` before the issue number so that the issue gets autoclosed on merge like so: `fixes #23469` 🙂
